File: app/src/main/java/com/absinthe/kage/device/DeviceObserverImpl.java
package com.absinthe.kage.device;

import com.absinthe.kage.device.model.DeviceInfo;

public class DeviceObserverImpl implements IDeviceObserver {

    @Override
    public void onFindDevice(DeviceInfo deviceInfo) { }

    @Override
    public void onLostDevice(DeviceInfo deviceInfo) { }

    @Override
    public void onDeviceConnected(DeviceInfo deviceInfo) { }

    @Override
    public void onDeviceDisConnect(DeviceInfo deviceInfo) { }

    @Override
    public void onDeviceConnectFailed(DeviceInfo deviceInfo, int errorCode, String errorMessage) { }

    @Override
    public void onDeviceInfoChanged(DeviceInfo deviceInfo) { }

    @Override
    public void onDeviceNotice(DeviceInfo deviceInfo) { }

    @Override
    public void onDeviceConnecting(DeviceInfo deviceInfo) { }
}

File: app/src/main/kotlin/com/absinthe/kage/device/DeviceObserverImpl.kt
package com.absinthe.kage.device

import com.absinthe.kage.device.model.DeviceInfo

open class DeviceObserverImpl : IDeviceObserver {
    override fun onFindDevice(deviceInfo: DeviceInfo) {}
    override fun onLostDevice(deviceInfo: DeviceInfo) {}
    override fun onDeviceConnected(deviceInfo: DeviceInfo) {}
    override fun onDeviceDisConnect(deviceInfo: DeviceInfo) {}
    override fun onDeviceConnectFailed(deviceInfo: DeviceInfo, errorCode: Int, errorMessage: String?) {}
    override fun onDeviceInfoChanged(deviceInfo: DeviceInfo) {}
    override fun onDeviceNotice(deviceInfo: DeviceInfo) {}
    override fun onDeviceConnecting(deviceInfo: DeviceInfo) {}
}